.github/workflows/main.yml

```diff
@@ -3,18 +3,32 @@ on:
     types: [labeled]
 
 jobs:
-  trivial:
+  base:
     runs-on: ubuntu-latest
     if: contains(github.event.pull_request.labels.*.name, 'ci:run')
-    name: First Pass Trigger Testing
+    name: Main Tests
     steps:
       - name: Checkout Repo
         uses: actions/checkout@v2
-      - name: Show Workspace Context
-        run: echo 'Workspace is ${{ github.workspace }}'
-      - name: Nasty Hack To Get Right Docker Build Context
-        run: cp ./.github/actions/docker_action/Dockerfile . && cp ./.github/actions/docker_action/action.yml .
-      - name: Run Docker Tests
-        uses: ./
-      - name: Show commit
-        run: git show -q
+      - name: test_all.sh running from Dockerfile.micro
+        uses: docker://jpwithers/tflite-micro-tests
+#       uncomment below to run only the five minute test bit.
+#        with:
+#          args: /bin/sh -c "/opt/tflm/tensorflow/lite/micro/tools/ci_build/test_all.sh"
+      - name: add a file for testing
+        run: git show -q > .commitinfo
+      - name: Commiting CI Variables Files
+        uses: EndBug/add-and-commit@3276c159cee23546e7fef9c8f9997e923587b543
```

Comment:
this is ok to use but let's start adding some notes on what 3rd party github actions we are using (in addition to the non-trivial things that the CI system is doing). I think adding a top-level continuous_integration.md and putting notes in there would be very useful. We can clean it up later but documenting as we go along would be really nice.